app.py

```diff
@@ -22,7 +22,102 @@ def init_db():
         FOREIGN KEY(title) REFERENCES notes(title)
     )""")
     
+    conn.close()
     
 init_db()
+
+###CREATION AND DELETION FUNCTIONS
+
+#Function to create a note
+def create_note(): 
+    pass 
+
+#function to delete a note
+def delete_note():
+    pass
+
+#function to create a tag
+def create_tag(): 
+    pass
+
+#function to delete a tag
+def delete_tag():
+    pass 
+
+#function to add content
+#I do not know if we need a separate function for edit content. I can explain the reasoning in class 
+def add_content(): 
+    pass 
+
+
+###SEARCHING FUNCTIONS
```

Comment:
respectfully i do not approve of this design! 
personally i don't think we are obeying the Don't Repeat Yourself (DRY) design paradigm. search does the same thing in each case: it takes a note field to search by, a query string, and a set of note fields to return that match the query. i think search should be one function, where our API supports 3 parameters in the GET payload: search_field, query, return_fields. the "search_field" argument that can be either "modified_date," "title", "created_date", and "tag." the query would be the string to search for, and the return_fields would be the aspects of a note to return. we could very flexibly do boolean logic if we wrote it like that, and we could easily extend our database to have more note metadata that way.